### PR TITLE
add routes-user branch for user

### PR DESCRIPTION
**Pull request by HammamAbuShehadeh**
#10 add user route

### CODE REVIEW

**Comment by HammamAbuShehadeh** [comment]
Done Update scheme models files

File: server/server/routes/userRoute.js
Comment (HammamAbuShehadeh) [inline server/server/routes/userRoute.js:1]: don't forget Auth after done

File: server/server/routes/userRoute.js [hunk 1]
```diff
@@ -1 +1,17 @@
...
+const {
+  ensureGuest,
+  ensureAuth,
+  verifyIsAdmin,
+} = require("../middlewares/userAuth");
+
+// get past completed orders ok
+router.get(
+  "/orders/past",
+  //  isAuthenticated,
```
Comment (HammamAbuShehadeh) [inline server/server/routes/userRoute.js:13]: Auth ?!

File: server/server/routes/userRoute.js [hunk 2]
```diff
@@ -14,4 +14,11 @@ router.get(
   userController.getPastOrders
 );
 
+// Get current active orders ok
+router.get(
+  "/orders/current",
+  //  isAuthenticated,
```
Comment (HammamAbuShehadeh) [inline server/server/routes/userRoute.js:20]: is for order in waiting only ?
Auth ?!

File: server/server/routes/userRoute.js [hunk 3]
```diff
@@ -14,11 +14,20 @@ router.get(
...
   "/orders/current",
   //  isAuthenticated,
   userController.getCurrentOrders
 );
 
+// Cancel an order 
+router.patch(
+  "/orders/:orderId/cancel",
+  //  isAuthenticated,
+  //  isOrderOwner,
```
Comment (HammamAbuShehadeh) [inline server/server/routes/userRoute.js:28]: Auth ?!

File: server/server/routes/userRoute.js [hunk 4]
```diff
@@ -14,20 +14,24 @@ router.get(
...
 router.patch(
   "/orders/:orderId/cancel",
   //  isAuthenticated,
   //  isOrderOwner,
   userController.cancelOrder
 );
 
+// get all dishes for user ok
+router.get("/Dishes", userController.getDishItems);
+
```
Comment (HammamAbuShehadeh) [inline server/server/routes/userRoute.js:34]: looks good thx

File: server/server/routes/userRoute.js [hunk 5]
```diff
@@ -32,6 +32,9 @@ router.patch(
 // get all dishes for user ok
 router.get("/Dishes", userController.getDishItems);
 
+// search for dish name min max price and catogire ok
+router.get("/Dishes/search", userController.searchDishes);
+
```
Comment (HammamAbuShehadeh) [inline server/server/routes/userRoute.js:37]: looks good thx

File: server/server/routes/userRoute.js [hunk 6]
```diff
@@ -35,6 +35,8 @@ router.get("/Dishes", userController.getDishItems);
 // search for dish name min max price and catogire ok
 router.get("/Dishes/search", userController.searchDishes);
 
+// detials for one dish for dynamic id in react ok
+router.get("/Dishes/:itemId", userController.getDishById);
```
Comment (HammamAbuShehadeh) [inline server/server/routes/userRoute.js:39]: looks good thx

File: server/server/routes/userRoute.js [hunk 7]
```diff
@@ -38,5 +38,8 @@ router.get("/Dishes/search", userController.searchDishes);
 // detials for one dish for dynamic id in react ok
 router.get("/Dishes/:itemId", userController.getDishById);
 
+//Get all Dishes in specific Restaurant ok
+router.get("/Restaurants/:restaurantID", userController.getDishesByRestaurant);
```
Comment (HammamAbuShehadeh) [inline server/server/routes/userRoute.js:42]: good work thx

File: server/server/routes/userRoute.js [hunk 8]
```diff
@@ -41,5 +41,15 @@ router.get("/Dishes/:itemId", userController.getDishById);
 //Get all Dishes in specific Restaurant ok
 router.get("/Restaurants/:restaurantID", userController.getDishesByRestaurant);
 
+// Add Dish to Cart ok
+router.post("/cart", userController.addToCart);
```
Comment (HammamAbuShehadeh) [inline server/server/routes/userRoute.js:45]: looks good thx

File: server/server/routes/userRoute.js [hunk 9]
```diff
@@ -41,5 +41,15 @@ router.get("/Dishes/:itemId", userController.getDishById);
...
+
+// signin & signout without google
+// sign in ok
+router.post("/signin", ensureGuest, userController.signin);
+
+// sign out ok
+router.get("/signout", userController.signout);
+
```
Comment (HammamAbuShehadeh) [inline server/server/routes/userRoute.js:53]: great feature

File: server/server/routes/userRoute.js [hunk 10]
```diff
@@ -51,5 +51,12 @@ router.post("/signin", ensureGuest, userController.signin);
 // sign out ok
 router.get("/signout", userController.signout);
 
+// get one user profile ok
+router.get(
+  "/:id",
+  //  ensureAuth,
```
Comment (HammamAbuShehadeh) [inline server/server/routes/userRoute.js:57]: Auth ?

File: server/server/routes/userRoute.js [hunk 11]
```diff
@@ -58,5 +58,8 @@ router.get(
   userController.getUserProfile
 );
 
+// add new user ok
+router.post("/newUser", ensureGuest, userController.addNewUser);
+
```
Comment (HammamAbuShehadeh) [inline server/server/routes/userRoute.js:63]: nice

File: server/server/routes/userRoute.js [hunk 12]
```diff
@@ -61,5 +61,12 @@ router.get(
 // add new user ok
 router.post("/newUser", ensureGuest, userController.addNewUser);
 
+// update user info ok
+router.put(
+  "/updateInfo/:id",
+  // ensureAuth,
```
Comment (HammamAbuShehadeh) [inline server/server/routes/userRoute.js:67]: nice one 
need Auth ?

File: server/server/routes/userRoute.js [hunk 13]
```diff
@@ -68,5 +68,19 @@ router.put(
...
+  userController.getUsersData
+);
+router.get("/allTokens", userController.getTokens);
+router.delete("/delete", userController.removeUser);
+router.delete("/deleteTokens", userController.removeTokens);
+
+// Check if the user not auth to login with google
+router.get("/google", ensureGuest, (_, res) => {
+  res.send('<a href="/auth/google">Authentication with Google</a>');
```
Comment (HammamAbuShehadeh) [inline server/server/routes/userRoute.js:83]: nice work 
i think need to remove line of a tags 
thx

**Review by HammamAbuShehadeh — COMMENTED** [review]
Looks good i think need Auth when test the code together
thx

File: server/server/routes/userRoute.js
Comment (MonaAlHajEid) [reply server/server/routes/userRoute.js:1]: ok

File: server/server/routes/userRoute.js (hunk 1 above)
Comment (MonaAlHajEid) [reply server/server/routes/userRoute.js:13]: sure after merge it

File: server/server/routes/userRoute.js (hunk 2 above)
Comment (MonaAlHajEid) [reply server/server/routes/userRoute.js:20]: yes 
sure after merge it

File: server/server/routes/userRoute.js (hunk 4 above)
Comment (MonaAlHajEid) [reply server/server/routes/userRoute.js:34]: y r welcome

File: server/server/routes/userRoute.js (hunk 6 above)
Comment (MonaAlHajEid) [reply server/server/routes/userRoute.js:39]: thx

File: server/server/routes/userRoute.js (hunk 7 above)
Comment (MonaAlHajEid) [reply server/server/routes/userRoute.js:42]: thx

File: server/server/routes/userRoute.js (hunk 9 above)
Comment (MonaAlHajEid) [reply server/server/routes/userRoute.js:53]: thx

File: server/server/routes/userRoute.js (hunk 11 above)
Comment (MonaAlHajEid) [reply server/server/routes/userRoute.js:63]: thanks

File: server/server/routes/userRoute.js (hunk 12 above)
Comment (MonaAlHajEid) [reply server/server/routes/userRoute.js:67]: thx 
sure after merge it

File: server/server/routes/userRoute.js (hunk 13 above)
Comment (MonaAlHajEid) [reply server/server/routes/userRoute.js:83]: thanks alot ^^

**Pull request merged**